### ISSUE TITLE
Hidden files are not included into globs like `**/*` inside hidden directories

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#8712]
Hidden files are not included into globs like `**/*` inside hidden directories.

Looks like an edge-case regression of #4832 and #5181.

I saw hidden directories excluded by default after #1656, so I've included in `AllCops` via glob. Maybe there is a conflict between them.

--------

## Expected behavior

Hidden files are included into globs like `**/*` if a hidden parent directory was included for cops.

## Actual behavior

With globs like `**/*` in RuboCop config hidden files (names starting with `.`) ignored in hidden directories even after including them for cops.

## Steps to reproduce the problem

```ruby
# Gemfile

# frozen_string_literal: true

gem 'rubocop', '0.90'
```

```yaml
# .rubocop.yml

inherit_mode:
  merge:
    - Include
    - Exclude

Metrics/BlockLength:
  Max: 2
  Exclude:
    - .toys/**/*

AllCops:
  NewCops: enable
  Include:
    - '.toys/**/{.,}*'
```

```ruby
# .toys/.test.rb

# frozen_string_literal: true

foo do
  bar
  baz
  qux
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.90.0 (using Parser 2.7.1.4, rubocop-ast 0.4.0, running on ruby 2.7.1 x86_64-linux)
```

**Comment by tleish** [comment]
@AlexWayfer - Based on the specs, it appears you should not be selecting a hidden file using `*`. 

```ruby
# spec/rubocop/path_util_spec.rb

describe '#match_path?', :isolated_environment do
  #...

    before do
      
      create_empty_file('dir/.hidden/file')
      create_empty_file('dir/.hidden_file')
    end

    it 'matches glob expressions' do
      #...

      expect(described_class.match_path?('**/*', 'dir/.hidden/file')).to be(false)
      expect(described_class.match_path?('**/*', 'dir/.hidden_file')).to be(true)
      expect(described_class.match_path?('**/.*/*', 'dir/.hidden/file')).to be(true)
      expect(described_class.match_path?('**/.*', 'dir/.hidden_file')).to be(true)

      #...
    end
    #...
  end
```

Instead, try the following changes to .rubocop.yml.

```diff
# .rubocop.yml

inherit_mode:
  merge:
    - Include
    - Exclude

Metrics/BlockLength:
  Max: 2
  Exclude:
+   - .toys/.*
    - .toys/**/*

AllCops:
  NewCops: enable
  Include:
    - '.toys/**/{.,}*'

```

**Comment by rrosenblum** [comment]
I think this could be solved by using `**/**`. Globbing with `*` does not include hidden files, however `**` will cover hidden files.

I have not tested this so I don't guarantee that it will work
```yaml
Metrics/BlockLength:
  Max: 2
  Exclude:
    - .toys/**/**
```

**Comment by bbatsov** [comment]
Closing as stale — niche glob edge case with hidden directories from 2020, no engagement in 5+ years. If still reproducible, please reopen.
